File: w09/d03/quin_cogdell/mail_objects_app/spec/javascripts/mailbox_spec.js
// Mailbox Object

// Mailbox should have an array of MailItem objects
// Mailbox should have an acceptMail method
// acceptMail should take a postcard or package
// acceptMail should place that mail in the Mailbox's array of MailItem objects
// Mailbox should have an openMail method that removes and returns the first item in its array of MailItem objects
// Mailbox should have a toString method that includes the number of postcards and packages in the mailbox
// Example: The mailbox has 16 postcards and 3 packages

describe("a Mailbox", function() {

  beforeEach(function(){
    mailbox = new Mailbox();
    mail1 = new mailItem('times square');
    mail2 = new mailItem('123 north pole');
    mailbox.acceptMail(mail1);
    mailbox.acceptMail(mail2);

  });

  it("has an array of MailItem Objects", function() {
    expect(mailbox.getMail()).toEqual([mail1, mail2]);
  });




});

File: w09/d01/edward_shin/mail_it/spec/mailbox_spec.rb
require_relative 'spec_helper'
require_relative '../mailbox'
require_relative '../postcard'
require_relative '../package'


describe Mailbox do
  it "should have an array of MailItem objects" do
    mailbox = Mailbox.new
    expect(mailbox.check_mail.class).to eq(Array)
  end

  it "should have an accept_mail method" do
    mailbox = Mailbox.new
    package = Package.new
    mailbox.accept_mail(package)
    expect(mailbox.check_mail.length).to eq(1)
    postcard = Postcard.new
    mailbox.accept_mail(postcard)
    expect(mailbox.check_mail.length).to eq(2)
  end

  it "should have an open_mail method that removes and returns the item" do
    mailbox = Mailbox.new
    package = Package.new
    postcard = Postcard.new
    mailbox.accept_mail(postcard)
    mailbox.accept_mail(package)
    expect(mailbox.open_mail).to eq(postcard)
  end

  it "should have a to_s method that returns the num of postcards and packages" do
    mailbox = Mailbox.new
    5.times do |n|
      n = Package.new
      mailbox.accept_mail(n)
    end
    7.times do |p|
    p = Postcard.new
      mailbox.accept_mail(p)
    end
    expect(mailbox.to_s).to eq("the mailbox has 5 packages and 7 postcards")
  end

end